spec/Section 7 -- Response.md

```diff
@@ -58,6 +58,10 @@ present in the response.
 If an error was raised during the execution that prevented a valid response, the
 `data` entry in the response should be `null`.
 
+Note: Request errors (including those raised during {ExecuteRequest()}) occur
+before execution begins; when a request error is raised the `data` entry should
```

Comment:
This is very hard for my brain to process. I may already be in `ExecuteRequest()` but still before execution begins?This feels weird.

Do we have a formal definition of what "execution" is and when it starts that we could link to maybe?

Reply:
That's what the purpose of this PR is :100:

Execution begins the moment that the `data` property should be added to the response, which is when the root selection set is executed (after validating variables, establishing a stream in the case of subscriptions, etc).

Reply:
I've taken a different tact by defining execution, see what you think.

Reply:
```
 We define _execution_ as the process of executing the operation's root
selection set through {ExecuteSelectionSet()}, thus _execution_ begins when
{ExecuteSelectionSet()} is called for the first time in a request. The
{ExecuteRequest()} algorithm is a preamble for _execution_.
```

This is clarifying but also also adding more language to processs. I think a rename of `ExecuteRequest()` would be even more clarifying.

Reply:
Why not both?

Reply:
Agreed.